### PR TITLE
Make MapVector from std container

### PR DESCRIPTION
**Pull request by Magoja**
Summary:
# Problem
We have `arrayVector()` to create an `ArrayVector` from `std::vector`. I'd like to have it something similar for `MapVector`

# Solution
I added `mapVector` that accepts `std::map` to create a test data without null values.

Differential Revision: D34281300

### CODE REVIEW

File: velox/vector/tests/VectorMaker.h [hunk 1]
```diff
@@ -587,6 +587,51 @@ class VectorMaker {
...
+    // Count number of elements.
+    vector_size_t numElements = 0;
+    for (const auto& map : data) {
+      numElements += map.size();
+    }
+
+    // Create the underlying flat vector.
+    auto keyVector = std::dynamic_pointer_cast<FlatVector<TKey>>(
+        BaseVector::create(CppToType<TKey>::create(), numElements, pool_));
+    auto valueVector = std::dynamic_pointer_cast<FlatVector<TValue>>(
```
Comment (kgpai) [inline velox/vector/tests/VectorMaker.h:608]: This only works if the value types are primitive.

File: velox/vector/tests/VectorMaker.h [hunk 2]
```diff
@@ -587,6 +587,51 @@ class VectorMaker {
         BaseVector::countNulls(nulls, 0, size));
   }
 
+  template <typename TKey, typename TValue>
+  MapVectorPtr mapVector(const std::vector<std::map<TKey, TValue>>& data) {
```
Comment (kgpai) [inline velox/vector/tests/VectorMaker.h:591]: It would be nice to support null values, null maps etc.

**Review by kgpai — COMMENTED** [review]
Thanks for doing this; I do believe that this function can actually be written to use 
```
MapVectorPtr mapVector(
      vector_size_t size,
      std::function<vector_size_t(vector_size_t /* row */)> sizeAt,
      std::function<TKey(vector_size_t /* idx */)> keyAt,
      std::function<TValue(vector_size_t /* idx */)> valueAt,
      std::function<bool(vector_size_t /*row */)> isNullAt = nullptr,
      std::function<bool(vector_size_t /*row */)> valueIsNullAt = nullptr)
  ```
  by creating a closure of the std::map in a std::function and passing to above function.

**Comment by mbasmanova** [comment]
Closing stale PRs. Please, reopen if you'd like to continue working on this PR.

**Pull request closed without merging**